File: task-3/prime.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

// Function to check if a number is prime
func isPrime(num int) bool {
	if num <= 1 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(num))); i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

// Main function
func main() {
	// Input from the user
	fmt.Print("Enter a number: ")
	var input string
	fmt.Scanln(&input)

	// Convert input to integer
	n, err := strconv.Atoi(input)
	if err != nil {
		fmt.Println("Invalid input. Please enter a valid number.")
		return
	}

	// Check if the number is prime
	if isPrime(n) {
		fmt.Printf("%d is a prime number.\n", n)
	} else {
		fmt.Printf("%d is not a prime number.\n", n)
	}
}

File: task-3/prime.rs
use std::io;

// Function to check if a number is prime
fn is_prime(num: u32) -> bool {
    if num <= 1 {
        return false;
    }
    for i in 2..=f64::sqrt(num as f64) as u32 {
        if num % i == 0 {
            return false;
        }
    }
    true
}

// Main function
fn main() {
    // Input from the user
    println!("Enter a number: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");

    // Convert input to unsigned integer
    let n: u32 = match input.trim().parse() {
        Ok(num) => num,
        Err(_) => {
            println!("Invalid input. Please enter a valid number.");
            return;
        }
    };

    // Check if the number is prime
    if is_prime(n) {
        println!("{} is a prime number.", n);
    } else {
        println!("{} is not a prime number.", n);
    }
}

